fearless_simd/examples/sigmoid.rs

```diff
@@ -1,6 +1,12 @@
 // Copyright 2024 the Fearless_SIMD Authors
 // SPDX-License-Identifier: Apache-2.0 OR MIT
 
+#![expect(
+    missing_docs,
+    clippy::shadow_unrelated,
```

Comment:
Apologies I didn't express this directly; we should also update the lint set to version 6.1 (https://linebender.org/wiki/canonical-lints/). The only relevant difference in that version is that it removes `shadow_unrelated` (see https://github.com/linebender/linebender.github.io/pull/90)

Luckily, we're already on v5, so this update only involves removing the expects for shadow_unrelated.